pubnub-core/src/data/channel/wildcard_spec.rs

```diff
@@ -0,0 +1,192 @@
+use crate::data::channel;
+use std::convert::TryFrom;
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+/// A wildcard Channel specifier.
+///
+/// This type represents a value suitable for use with [wildcard subscribe]
+/// feature.
+///
+/// Currently you can have up to three levels deep with your channel segment
+/// hierarchies, `a.b.c`, for example.
+///
+/// Developers MUST use .* to denote a wildcard subscription. Just having a * at
+/// the end of channel names will NOT get translated to .* and will result in a
+/// subscribe error.
```

Comment:
Does it really return an error?

Reply:
It's just copied from the docs: https://support.pubnub.com/support/solutions/articles/14000043663-how-do-i-subscribe-to-a-wildcard-channel-